### DISCUSSION TITLE
Tacotron 2 fp16

### DISCUSSION THREAD

**Discussion by JRMeyer** [coqui-ai/TTS#318] [Archived]
<USER>>>> vcjobacc<br/> <DATE> [December 6, 2020, 11:45am]<br/> <br/> Hello everyone! *slash* <br/> Since I've got an RTX card, I can potentially do fp16 training with amp,<br/> right? *slash* <br/> The question is, has anyone tried it already? Just wanna know if it's<br/> worth to try. Can I start training with fp16 and, let's say, on 100k<br/> steps get back to fp32? *slash* <br/> Thank you!<br/> <br/> [This is an archived TTS discussion thread from discourse.mozilla.org/t/tacotron-2-fp16]<br/>

**Comment by JRMeyer** [comment]
<USER>>>> erogol<br/> <DATE> [December 7, 2020, 9:47am]<br/> <br/> most of the training code already supports f16 training. You just need<br/> to set `apex_amp_level='O1'` or `mixed_precision=true` in config file<br/> depending on the training script using apex or pytorch<br/> <br/> [Archived Post]<br/>

**Comment by JRMeyer** [comment]
<USER>>>> vcjobacc<br/> <DATE> [December 11, 2020, 8:30pm]<br/> <br/> Thank you! *slash* <br/> What's your own experience with *mixed_precision=true*? *slash* <br/> I have multiple tries, one of them finished with 'Zero loss & Inf<br/> gradients' all of a sudden after like 16th epoch, and the other for some<br/> reasons dropped the quality suddenly with the same number of frames per<br/> second (see pic.). Has it something to deal with AMP? Thank you!<br/> <br/> [Archived Post]<br/>

**Comment by JRMeyer** [comment]
<USER>>>> synesthesiam<br/> <DATE> [December 12, 2020, 7:26pm]<br/> <br/> > I have multiple tries, one of them finished with 'Zero loss & Inf<br/> > gradients'<br/> <br/> I get the same thing with CUDA 11 and a 2080 Ti. Tried PyTorch 1.6 and<br/> 1.7.<br/> <br/> [Archived Post]<br/>

**Comment by JRMeyer** [comment]
<USER>>>> erogol<br/> <DATE> [December 12, 2020, 10:42pm]<br/> <br/> they release 1.7.1 mentioning some perf issues in 1.7<br/> <br/> Maybe useing the lates pytorch version moght solve. otherwise it is<br/> finding the right LR that works .<br/> <br/> [Archived Post]<br/>
